tck/optional/src/main/java/org/eclipse/microprofile/metrics/test/optional/MetricAppBeanOptional.java

```diff
@@ -0,0 +1,283 @@
+        return "This is a GET request with a List3";
+    }
+    
+    @GET
+    @Path("/get-vararg-param1")
+    public String  getVarargParam1(@QueryParam("qp1") Boolean... v1) throws Exception {
+        return "This is a GET request with vararg parameter1";
+    }
+    
+    @GET
+    @Path("/get-vararg-param2")
+    public String  getVarargParam2(@QueryParam("qp1") int v1, @QueryParam("qp2") String... v2) throws Exception {
+        return "This is a GET request with vararg parameter2";
+    }
+    
+    @GET
+    @Path("/get-array-param1")
+    public String  getArrayParam1(@QueryParam("qp1") String[] v1) throws Exception {
+        return "This is a GET request with array parameter";
+    }
+    
+    @GET
+    @Path("/get-array-param2")
+    public String  getArrayParam2(@QueryParam("qp1") int[] v1) throws Exception {
+        return "This is a GET request with array parameter";
+    }
+    
+    @GET
+    @Path("/get-array-param3")
+    public String  getArrayParam3(@QueryParam("qp1") Double[] v1) throws Exception {
+        return "This is a GET request with array parameter";
+    }
+
+    
+    @GET
+    @Path("/get-multiple-param1")
+    public String  getMultipleParam1(
+            @QueryParam("qp1") boolean v1,
+            @QueryParam("qp2") int v2,
+            @QueryParam("qp3") double v3,
+            @QueryParam("qp4") String v4,
+            @QueryParam("qp5") long v5) throws Exception {
+        return "This is a GET request with multiple parameters1";
+    }
+    
+    @GET
+    @Path("/get-multiple-param2")
+    public String  getMultipleParam2(@QueryParam("qp1") String v1, @QueryParam("qp2") List<String> v2) throws Exception {
+        return "This is a GET request with multiple parameters2";
+    }
+    
+    @GET
+    @Path("/get-multiple-param3")
+    public String  getMultipleParam3(@QueryParam("qp1") boolean v1,
+            @QueryParam("qp2") Boolean v2, @QueryParam("qp3") double v3, @QueryParam("qp4")String...v4) throws Exception {
+        return "This is a GET request with multiple parameters3";
+    }
+    
+    @GET
+    @Path("/get-multiple-param4")
+    public String  getMultipleParam4(@QueryParam("qp1") Set<String> v1,
+            @QueryParam("qp2") SortedSet<Integer> v2, @QueryParam("qp3") double[] v3) throws Exception {
+        return "This is a GET request with multiple parameters4";
+    }
+    
+    
+    @GET
+    @Path("/get-name-object")
+    public String  getNameObject(@QueryParam("qp1") NameObject v1) throws Exception {
+        return "This is a GET request with NameObject";
+    }
+    
+    @GET
+    @Path("/get-async")
+    public void  getAsync(@Suspended final AsyncResponse ayncResponse) throws Exception {
+        Thread thread = new Thread ( () -> {
+            try {
+                Thread.sleep(5000);
+            } 
+            catch (Exception e) {
+                System.err.println(e.toString());
+            }
+        });
+        thread.run();
```

Comment:
- you need to call `thread.start()` not `run()` to actually start the thread
- `asyncResponse.resume` should be called in the async thread (after the `Thread.sleep` call), not in the main thread, otherwise the call is not really asynchronous

This works for me:

```
@GET
@Path("/get-async")
public void  getAsync(@Suspended final AsyncResponse asyncResponse) throws Exception {
    Thread thread = new Thread ( () -> {
        try {
            Thread.sleep(5000);
            asyncResponse.resume("This is a GET request with AsyncResponse");
        }
        catch (Exception e) {
            System.err.println(e.toString());
        }
    });
    thread.start();
}
```

Reply:
fixed!